source/AI.cpp

```diff
@@ -829,7 +829,9 @@ void AI::Step(const PlayerInfo &player, Command &activeCommands)
 				else
 					parent.reset();
 
-				it->SetParent(parent);
+				// Only set a new parent if the new parent can legitimately carry the fighter
+				if(!it->GetParent() || !parent || !parent->GetParent() || parent->CanCarry(*it))
```

Comment:
why `|| !parent->GetParent()`? shouldn't it be an `&&`

Reply:
This is required for the abandoned fighter to keep station with player flagship.

With out this, you get the following: strange fighter behavior bouncing around to many escorts that can't carry the ship.

This enables the fighter to finally resolve to the player flagship carrier (which it can't board) when there are no free bays available for the fighter to board amongst the fleet.

This is the final fallback behavior in the "testing done" section.

Reply:
I can add a code comment for clarity around this so it is obvious post-merge.

Reply:
I think this is resolved with the rewrite.